src/blockdata/script.rs

```diff
@@ -556,10 +551,58 @@ impl<'a> Iterator for Instructions<'a> {
 
 impl Builder {
     /// Creates a new empty script
-    pub fn new() -> Builder {
+    pub fn new() -> Self {
         Builder(vec![], None)
     }
 
+    /// Generates P2PK-type of scriptPubkey
+    pub fn gen_p2pk(pubkey: &PublicKey) -> Self {
+        Builder::new().push_key(pubkey)
+            .push_opcode(opcodes::all::OP_CHECKSIG)
+    }
+
+    /// Generates P2PKH-type of scriptPubkey
+    pub fn gen_p2pkh(pubkey_hash: &PubkeyHash) -> Self {
+        Builder::new().push_opcode(opcodes::all::OP_DUP)
+            .push_opcode(opcodes::all::OP_HASH160)
+            .push_slice(&pubkey_hash[..])
+            .push_opcode(opcodes::all::OP_EQUALVERIFY)
+            .push_opcode(opcodes::all::OP_CHECKSIG)
+    }
+
+    /// Generates P2SH-type of scriptPubkey with a given hash of the redeem script
+    pub fn gen_p2sh(script_hash: &ScriptHash) -> Self {
+        Builder::new().push_opcode(opcodes::all::OP_HASH160)
+            .push_slice(&script_hash[..])
+            .push_opcode(opcodes::all::OP_EQUAL)
+    }
+
+    /// Generates P2WPKH-type of scriptPubkey
+    pub fn gen_v0_p2wpkh(pubkey_hash: &WPubkeyHash) -> Self {
+        Self::gen_witness(bech32::u5::try_from_u8(0).unwrap(), &pubkey_hash.to_vec())
+    }
+
+    /// Generates P2WSH-type of scriptPubkey with a given hash of the redeem script
+    pub fn gen_v0_p2wsh(script_hash: &WScriptHash) -> Self {
+        Self::gen_witness(bech32::u5::try_from_u8(0).unwrap(), &script_hash.to_vec())
+    }
+
+    /// Generates P2WSH-type of scriptPubkey with a given hash of the redeem script
+    pub fn gen_witness(ver: bech32::u5, program: &Vec<u8>) -> Self {
```

Comment:
`gen_witness_program` I think is a better name. You're not generating a witness, witnesses are inputs.